perf/lock.ts

```diff
@@ -0,0 +1,93 @@
+import {assert} from './replicache.js';
+import {newTab} from '../src/test-util';
+import {uuid} from '../src/sync/uuid.js';
+
+enum Mode {
+  SameTab = '',
+  OtherTab = 'other tab',
+  OtherProcess = 'other process',
+}
+
+export function benchmarks() {
+  return [
+    benchmarkUncontendedRead(),
+    benchmarkHeldRead(Mode.SameTab),
+    benchmarkHeldRead(Mode.OtherTab),
+    benchmarkHeldRead(Mode.OtherProcess),
+    benchmarkUncontendedWrite(),
+  ];
+}
+
+export async function isHeld(name: string) {
+  return await navigator.locks.request(
+    name,
+    {ifAvailable: true},
+    async lock => lock === null,
+  );
+}
+
+export function lockShared(name: string) {
+  const p = new Promise(res => {
+    window['resolve'] = res;
```

Comment:
maybe use a less generic name?

Reply:
Changed to use a map by lock name, since once ESLint started actually checking this file it really didn't like these property assignments to `window`. Resolves this.